reference/2_constructing_data/2_1_setters_deleters/README.md

```diff
@@ -0,0 +1,90 @@
+# Using setters & deleters
+
+Setters are the atomic data manipulation operations in Cranq. There are built-in nodes for dictionaries/complex types, and one for arrays.
+
+## Example - Adding/assigning items to dictionaries
+
+> **_Try out:_**
+>
+> Place node **tutorials/data_constr/setter_dict** or **[download .cranqj](cranqj/data_constr_setter_dict.cranqj)** 
+
+Adding/assigning items in a dictionary can be performed with the ```data/dictionary/Item setter``` node
+- It takes a dictionary, and a key/value pair as input
+- If the key already exists in it, the value will be overwritten
+- If the key does not exist, it will be created
+
+![](images/2021-07-19-13-34-57.png)
+
+### Sample output:
+
+```json
+{
+  "key2": "value2",
+  "key1": "value1"
+}
+```
+
+## Example - Inserting items into arrays
+
+> **_Try out:_**
+>
+> Place node **tutorials/data_constr/setter_arr** or **[download .cranqj](cranqj/data_constr_setter_arr.cranqj)** 
+
+Adding/assigning elements in arrays can be performed with the ```data/array/Item inserter``` node.
+- It takes the target array, the item to add, and the index to insert at as input
+- Over-indexing is possible, but the item will be added to the end of the array

```

Comment:
Is this a bug or feature?

Reply:
Will re-phrase as `Over-indexing is possible, but items can only be added to the end of the array` - I think that's an intended behaviour